cognite/client/data_classes/user_profiles.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Sequence, cast
+
+from cognite.client.data_classes._base import CogniteResource, CogniteResourceList
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+class UserProfile(CogniteResource):
+    """User profiles is an authoritative source of core user profile information (email, name, job title, etc.)
+    for principals based on data from the identity provider configured for the CDF project.
+
+    Args:
+        user_identifier (str | None): Uniquely identifies the principal the profile is associated with. This property is guaranteed to be immutable.
+        given_name (str | None): The user's first name.
+        surname (str | None): The user's last name.
+        email (str | None): The user's email address (if any). The email address is is returned directly from the identity provider and not guaranteed
+            to be verified. Note that the email is mutable and can be updated in the identity provider. It should not be used to uniquely identify as a
+            user. Use the user_identifier property instead.
+        display_name (str | None): The display name for the user.
+        job_title (str | None): The user's job title.
+        last_updated_time (int | None): The number of milliseconds since 00:00:00 Thursday, 1 January 1970, Coordinated Universal Time (UTC), minus leap seconds.
+        cognite_client (CogniteClient | None): No description.
+    """
+
+    def __init__(
+        self,
+        user_identifier: str | None = None,
```

Comment:
Looking at the docs, this is required so suggest making it required here as well. 
```suggestion
        user_identifier: str,
```

Reply:
Can't unfortunately, as it breaks how we load data into CogniteResource classes

Reply:
I.e. this monstrosity which I'm about to remove in an upcoming PR:
```py
elif isinstance(resource, dict):
    instance = cls(cognite_client=cognite_client)  # <--
    for key, value in resource.items():
        snake_case_key = to_snake_case(key)
        if hasattr(instance, snake_case_key):
            setattr(instance, snake_case_key, value)
    return instance
```